rs/nervous_system/common/src/memory_manager_upgrade_storage.rs

```diff
@@ -326,6 +327,9 @@ impl<'a, M: Memory> Buf for SizeAwareReader<'a, M> {
     }
 
     fn advance(&mut self, cnt: usize) {
```

Comment:
Seems like there's undesirable code duplication between here and rs/nervous_system/common/src/dfn_core_stable_mem_utils.rs

Is there a simple way to avoid that / a good reason to duplicate the code?

Reply:
True. Although, I think creating a common abstraction for both sides is difficult, mainly because they depend on slightly different StableMemory layers to read the memory. It might be possible to simply refactor some common logic to a few functions, but I imagine the reader would have to deal with 3 modules instead of 2, since there is still parts that cannot be shared but are non-trivial. I'm not convinced that would be better.

Reply:
Okay, let's not block on this, then